BubblesKotlin/app/src/main/java/com/example/android/bubbles/data/NotificationHelper.kt

```diff
@@ -61,16 +67,45 @@ class NotificationHelper(private val context: Context) {
                 }
             )
         }
+        updateShortcuts()
     }
 
     @WorkerThread
-    fun showNotification(chat: Chat, fromUser: Boolean) {
-        val icon = Icon.createWithAdaptiveBitmap(
-            BitmapFactory.decodeResource(
-                context.resources,
-                chat.contact.icon
+    fun updateShortcuts() {
+        val shortcuts = Contact.CONTACTS.map { contact ->
+            val icon = Icon.createWithAdaptiveBitmap(
+                context.resources.assets.open(contact.icon).use { input ->
+                    BitmapFactory.decodeStream(input)
+                }
             )
-        )
+            ShortcutInfo.Builder(context, contact.id.toString())
+                .setActivity(ComponentName(context, MainActivity::class.java))
```

Comment:
Add some comments here, just to explain what's going on

Reply:
Done